CLIENTS.md

```diff
@@ -0,0 +1,369 @@
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "publicKey": {
+                "rawBytes": "<base64 DER-encoded public key>"
+            },
+            // Must match signing algorithm
+            "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+        }
+    }
+}
+```
+
+The response will be a
+[`TransparencyLogEntry` message](https://github.com/sigstore/protobuf-specs/blob/5296f13d62e7fad428581d969f664c30cc52f549/protos/sigstore_rekor.proto#L94),
+which should be persisted in a bundle. Clients no longer need to transform the
+Rekor response into a `TLE` message to store in the bundle.
+
+### Two Entry Types
+
+Rekor v2 only supports `hashedrekord` (`HashedRekordRequestV0_0_2`) and
+`dsse` (`DSSERequestV0_0_2`) entry types, dropping a number of unused types
+such as `jar`, `alpine`, `rpm`, and the older types `rekord` and `intoto`.
+Additional types may be added in the future if there is demand, but this
+will require updating the client specification so that all clients implement
+support for these types.
+
+### Certificate and Public Key Verifiers
+
+Rekor v2 only supports signature verification using a certificate or a
+public key, dropping support for PGP, minisign, pkcs7, SSH and TUF.
+Additional verifiers may be added in the future, but this will also require
+updating the client specification.
+
+### Handling Longer Requests
+
+Clients need to increase request timeouts to at least 10 seconds.
+
+Rekor now batches uploads so that checkpoints are published less frequently.
+Additionally, we plan to support synchronous witnessing, where third-party
+witnesses independently verify the consistency of the log and Rekor provides
+co-signed checkpoints with each upload response.
+
+If a client needs to create multiple entries, it is recommended to upload those
+entries in parallel.
+
+## TrustedRoot lookup by checkpoint key ID rather than log ID
+
+Log ID, the SHA-256 digest of the log's public key, is used as a "unique"
+identifier for a log. The TransparencyLogEntry message includes it in a
+bundle, and clients use that log ID to lookup the correct log public key
+in the TrustedRoot to verify the bundle.
+
+Rekor will no longer include log IDs in the response, and instead clients
+should use the checkpoint key ID as specified in the
+[C2SP spec](https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures)
+to lookup the correct log public key to verify a bundle. Each transparency
+log instance in the TrustedRoot will include a `checkpoint_key_id` instead
+of a `log_id`.
+
+For more information, log IDs are not necessarily unique identifiers for
+a log, since a log may reuse its public key among instances. Additionally,
+the log origin is not necessarily a unique identifier, because multiple logs
+may be hosted by one origin. Even the combination of both is not necessarily
```

Comment:
Each shard should be uniquely named

Reply:
Thanks, added a mention that this SHOULD NOT happen. Technically, a deployment can reuse the key and/or origin, which the checkpoint key ID calculation takes into account.